test/test_ner_amp.py

```diff
@@ -0,0 +1,92 @@
+import numpy as np
+
+from farm.data_handler.data_silo import DataSilo
+from farm.data_handler.processor import NERProcessor
+from farm.modeling.optimization import initialize_optimizer
+from farm.infer import Inferencer
+from farm.modeling.adaptive_model import AdaptiveModel
+from farm.modeling.language_model import LanguageModel
+from farm.modeling.prediction_head import TokenClassificationHead
+from farm.modeling.tokenization import Tokenizer
+from farm.train import Trainer
+from farm.utils import set_all_seeds, initialize_device_settings
+
+import logging
+
+
+def test_ner(caplog):
+    caplog.set_level(logging.CRITICAL)
+
+    set_all_seeds(seed=42)
+    device, n_gpu = initialize_device_settings(use_cuda=True)
+    n_epochs = 1
+    batch_size = 2
+    evaluate_every = 1
+    lang_model = "bert-base-german-cased"
+
+    tokenizer = Tokenizer.load(
+        pretrained_model_name_or_path=lang_model, do_lower_case=False
+    )
+
+    ner_labels = ["[PAD]", "X", "O", "B-MISC", "I-MISC", "B-PER", "I-PER", "B-ORG", "I-ORG", "B-LOC", "I-LOC", "B-OTH",
+                  "I-OTH"]
+
+    processor = NERProcessor(
+        tokenizer=tokenizer,
+        max_seq_len=8,
+        data_dir="samples/ner",
+        train_filename="train-sample.txt",
+        dev_filename="dev-sample.txt",
+        test_filename=None,
+        delimiter=" ",
+        label_list=ner_labels,
+        metric="seq_f1"
+    )
+
+    data_silo = DataSilo(processor=processor, batch_size=batch_size)
+    language_model = LanguageModel.load(lang_model)
+    prediction_head = TokenClassificationHead(layer_dims=[768, len(ner_labels)])
+
+    model = AdaptiveModel(
+        language_model=language_model,
+        prediction_heads=[prediction_head],
+        embeds_dropout_prob=0.1,
+        lm_output_types=["per_token"],
+        device=device
+    )
+
+    model, optimizer, lr_schedule = initialize_optimizer(
+        model=model,
+        optim_opts={'name': 'AdamW', 'lr': 2E-05},
```

Comment:
As mentioned, it would be great if we can hide some of the complexity for unexperienced users by providing decent defaults. Repeating `optim_opts={'name': 'AdamW', 'lr': 2E-05}` in every example, tutorial, test etc. is also not super nice. 

What do you think about some signature like this: 
``` 
initialize_optimizer(model,
                         n_batches,
                         n_epochs,
                         lr,
                         optimizer_name="AdamW",
                         optimizer_opts=None,
                         schedule_name="WarmupLinearSchedule",
                         schedule_opts=None,
                         warmup_proportion=0.1,
                         use_amp=None,
                         grad_acc_steps=1,
                         local_rank=-1,
                         log_learning_rate=False):
``` 
I am also happy to tackle that afterwards, if you don't have time to work on this.